src/python/pants/backend/project_info/peek.py

```diff
@@ -0,0 +1,146 @@
+    RAW = "raw"
+    JSON = "json"
+
+
+class PeekSubsystem(GoalSubsystem):
+    """Display BUILD file info to the console.
+
+    In its most basic form, `peek` just prints the contents of a BUILD file. It can also display
+    multiple BUILD files, or render normalized target metadata as JSON for consumption by other
+    programs.
+    """
+
+    name = "peek"
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--output",
+            type=OutputOptions,
+            default=OutputOptions.RAW,
+            help="which output style peek should use",
+        )
+
+    @property
+    def output(self) -> OutputOptions:
+        return cast(OutputOptions, self.options.output)
+
+
+class Peek(Goal):
+    subsystem_cls = PeekSubsystem
+
+
+_T = TypeVar("_T")
+_U = TypeVar("_U")
+
+
+def _intersperse(x: _T, ys: Iterable[_U]) -> Iterable[Union[_T, _U]]:
+    # Use stateful iterator to safely iterate through first item
+    it = iter(ys)
+    for y in it:
+        yield y
+        break
+    for x, y in zip(repeat(x), it):
+        yield x
+        yield y
+
+
+def _render_raw(fc: FileContent, encoding: str = "utf-8") -> str:
+    dashes = "-" * len(fc.path)
+    content = fc.content.decode(encoding)
+    parts = [dashes, fc.path, dashes, content]
+    if not content.endswith(os.linesep):
+        parts.append("")
+    return os.linesep.join(parts)
+
+
+def _render_json(ts: Iterable[Target]) -> str:
+    data = dict(map(_target_to_kv, ts))
+    return json.dumps(data, indent=2, cls=_PeekJsonEncoder)
+
+
+def _target_to_kv(t: Target) -> Tuple[str, dict]:
+    nothing = object()
+    d = {
+        k.alias: v.value
+        for k, v in t.field_values.items()
+        # don't report default values in normalized output
+        if getattr(k, "default", nothing) != v.value
+    }
+    d["alias"] = t.alias
+    return t.address.spec, d
+
+
+class _PeekJsonEncoder(json.JSONEncoder):
+    """Allow us to serialize some commmonly-found types in BUILD files."""
+
+    safe_to_str_types = (Requirement,)
+
+    def default(self, o):
+        """Return a serializable object for o."""
+        if is_dataclass(o):
+            return asdict(o)
+        elif isinstance(o, Mapping):
+            return dict(o)
+        elif isinstance(o, self.safe_to_str_types):
+            return str(o)
+        else:
+            try:
+                return super().default(o)
+            except TypeError:
+                # punt and just try str()
+                # TODO: can we find a better strategy here?
+                return str(o)
+
+
+@goal_rule
+async def peek(
+    console: Console,
+    subsys: PeekSubsystem,
+    addresses: Addresses,
+) -> Peek:
+    # TODO: better options for target selection (globs, transitive, etc)?
+    targets: Iterable[Target]
+    # TODO: handle target not found exceptions better here
+    targets = await Get(Targets, Addresses, addresses)
```

Comment:
cc @stuhood on this to confirm.

`await Get(Targets, Addresses)` expands out all "file targets" every time. Meaning, it generates a more granular target which only has one source file, rather than the original "BUILD target" you see in the BUILD file. Run `./pants peek --output=json src/python/pants/util:util` - surprisingly, you'll get >1 target.

Almost certainly, we instead want to use `await Get(UnexpandedTargets, Addresses, addresses)`. This will only give back the precise target specified by the user, i.e. `src/python/pants/util:util` gives you that BUILD target, and then `src/python/pants/util/strutil.py` gives you the more granular file target `strutil.py`.

@stuhood can you think of any scenario here where we wouldn't want to use `UnexapandedTargets`?